test/e2e/e2e_test.go

```diff
@@ -690,6 +683,68 @@ func checkMetricsCollection(t *testing.T, promClient prometheusv1.API, namespace
 	}
 }
 
+func runPromQueryForVector(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (model.Vector, error) {
+	results, warnings, err := promClient.Query(context.Background(), query, sampleTime)
+	if err != nil {
+		return model.Vector{}, err
```

Comment:
nit: Please return nil. Best practice is to return the 'zero' value of return values when returning an error, and that value for `Vector` is `nil` due to it being an alias for a slice type.